packages/itwinui-react/src/core/utils/functions/dev.ts

```diff
@@ -2,14 +2,21 @@
  * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
  * See LICENSE.md in the project root for license terms and full copyright notice.
  *--------------------------------------------------------------------------------------------*/
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-const isJest = typeof (globalThis as any).jest !== 'undefined';
+const isTestingFramework = () => {
+  return (
+    typeof vitest !== 'undefined' ||
```

Comment:
I think it will also fail for vitest if [`globals`](https://vitest.dev/config/#globals) is not enabled.

Reply:
No, there is no difference between `vitest` and `globalThis.vitest`; they are one and the same. With `globals` disabled, both will be empty (you should verify this just to be sure) because the usage requires an explicit import ([docs](https://vitest.dev/api/vi.html)).

We might need a different way of detecting vitest.

Reply:
`process.env.VITEST` did not do what you wanted?
<img width="668" alt="image" src="https://github.com/iTwin/iTwinUI/assets/81580355/064e215f-5809-45c8-9766-813c5f6c9a15">


https://vitest.dev/config/#configuration

Reply:
`process.env` is not available in the browser, but we could use it with `try`-`catch`. e.g.:
```js
let isVitest = false;
try {
  isVitest = process.env.VITEST !== undefined;
} catch {}
```

I'm also thinking we could additionally check a custom environment variable that we can ask users to manually set in our documentation. e.g:
> By default, iTwinUI tries to detect `jest` and `vitest` to bail out of running code that's irrelevant for such test environments. If you are using a different test runner, such as `mocha`, please manually set the environment variable `IS_UNIT_TEST=TRUE`.

Reply:
Lets go with this for now:

```js
const isVitest = typeof globalThis.__vitest_index__ !== "undefined";
```

via [`console-fail-test/src/environments/vitest.ts`](https://github.com/JoshuaKGoldberg/console-fail-test/blob/main/src/environments/vitest.ts) (by one of mocha maintainers)

It's better than what we have right now (nothing). We can revisit later if it ever causes problems.